### PR TITLE
Polish up layered_test_check

### PR DESCRIPTION
**Pull request by nedbat**
Move method out of the class that could be useful for future thing.  Booleanify the return value just in case. Add test cases to get coverage to 100%

### CODE REVIEW

**Comment by nedbat** [comment]
@cpennington

File: edx_lint/pylint/layered_test_check.py [hunk 1]
```diff
@@ -32,32 +52,13 @@ class LayeredTestClassChecker(BaseChecker):
     }
 
     @utils.check_messages(MESSAGE_ID)
```
Comment (cpennington) [inline edx_lint/pylint/layered_test_check.py]: Do we need to leave this decorator on the `is_test_case_class` method?

Comment (nedbat) [inline edx_lint/pylint/layered_test_check.py]: No, in fact, that was one of the first things I changed.  That decorator is used on visit_ methods to limit the exploration. Not sure why we had it on a helper method originally.

Comment (nedbat) [inline edx_lint/pylint/layered_test_check.py]: I should not assume that silence is assent, correct?

**Pull request merged**